bootstrap/secret/insecure.go

```diff
@@ -146,3 +148,40 @@ func (p *InsecureProvider) ListSecretPaths() ([]string, error) {
 
 	return results, nil
 }
+
+// RegisteredSecretUpdatedCallback registers a callback for a secret.
+func (p *InsecureProvider) RegisteredSecretUpdatedCallback(path string, callback func(path string)) (func(), error) {
+	if _, ok := p.registeredSecretCallbacks[path]; ok {
+		return func() {
+			p.DeregisterSecretUpdatedCallback(path)
+		}, fmt.Errorf("there is a callback already registered for path '%v'", path)
+	}
+
+	// Register new call back for path.
+	p.registeredSecretCallbacks[path] = callback
+
+	return func() {
+		p.DeregisterSecretUpdatedCallback(path)
+	}, nil
+}
+
+// SecretUpdatedAtPath performs updates and callbacks for an updated secret or path.
+func (p *InsecureProvider) SecretUpdatedAtPath(path string) {
+	p.lastUpdated = time.Now()
+	if p.registeredSecretCallbacks != nil {
+		// Execute Callback for provided path.
+		for k, v := range p.registeredSecretCallbacks {
+			if k == path {
+				v(path)
+				p.lc.Debugf("callback registered for path: '%s'", path)
```

Comment:
```suggestion
				p.lc.Debugf("invoking callback registered for path: '%s'", path)
				v(path)
```


We need to know what the code was doing BEFORE it calls the callback